devito/ir/support/basic.py

```diff
@@ -426,7 +426,32 @@ def distance(self, other):
             else:
                 v = i - j
                 if v.is_Number and v.is_finite:
-                    return Vector(S.ImaginaryUnit)
+                    if i.is_Number and j.is_Number:
+                        return Vector(S.ImaginaryUnit)
+                    else:
+                        # If both i and j are not numbers,
+                        # there may be dimension-dependent dependencies
+                        # so we append the distance
+
+                        # For example:
```

Comment:
First of all, why in this example is the first Eq even needed? is the dependency along `y` across the u[1,1] write and the u[0, y+1] read?

Also, rewrite for homogeneity (and assuming what I wrote above is correct) as

```
# E.g., `self=W<f,[1, 1]>` and `other=R<f[0, y+1]>`
```

Reply:
The first equation is necessary since this `else` statement catches a dependency between writing into `[u0,y]` and reading from `u[0, y+1]` (we can' t schedule the two statements into the same loop because of this, or else the read from `u[0, y+1]` will read points not yet initialized by `u[0,y]`

Reply:
do I write it as `self=W<u,[0.y]` and `other=R<u,[0,y+1]>` then?

Reply:
And similarly for the example I gave about 10 lines below:
```
Eq(u[0,y], 1)
Eq(u[1,y+1], u[0,1])
```
I would write
`self=R<u,[1,2]> and other=W<u,[1,y+1]>` instead?

Reply:
right, yes